alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -85,59 +93,62 @@ void TeamManager::setTeamTimeout(AlicaTime t)
 
 void TeamManager::readSelfFromConfig(essentials::IdentifierConstPtr agentID)
 {
-    essentials::SystemConfig& sc = essentials::SystemConfig::getInstance();
+    YAML::Node config = _engine->getConfig();
     const std::string localAgentName = _engine->getLocalAgentName();
 
     if (agentID == nullptr) {
         constexpr auto notAValidID = std::numeric_limits<uint64_t>::max();
-        uint64_t id = sc["Local"]->tryGet<uint64_t>(notAValidID, "Local", "ID", NULL);
+        uint64_t id = config["Local"]["ID"].as<uint64_t>(notAValidID);
         if (id != notAValidID) {
             _localAnnouncement.senderID = _engine->getID(id);
         } else {
             _localAnnouncement.senderID = _engine->generateID(DEFAULT_AGENT_ID_SIZE);
             ALICA_DEBUG_MSG("TM: Auto generated id " << _localAnnouncement.senderID);
-            bool persistId = sc["Alica"]->tryGet<bool>(false, "Alica", "PersistID", NULL);
-            if (persistId) {
-                try{
-                    auto* configLocal = sc["Local"];
-                    configLocal->setCreateIfNotExistent(static_cast<uint64_t>(*_localAnnouncement.senderID), "Local", "ID", NULL);
-                    configLocal->store();
-                } catch(...) {
-                    ALICA_ERROR_MSG("TM: impossible to store ID " << _localAnnouncement.senderID);
-                }
-            }
         }
     } else {
-        _localAnnouncement.senderID = agentID;
+        if (!_localAgent) {
+            _localAnnouncement.senderID = agentID;
+        }
+    }
+
+    if (!_localAgent) {
```

Comment:
if(!_localAgent) is repeated too many times. For now we can add just this check at the entry of this method:
if (_localAgent)
  return;